internal/nearcache/repairing_task.go

```diff
@@ -0,0 +1,576 @@
+		}
+		key = data
+	}
+	return h.ps.GetPartitionID(key)
+}
+
+// MetaDataContainer contains one partitions' invalidation metadata.
+// port of: com.hazelcast.internal.nearcache.impl.invalidation.MetaDataContainer
+type MetaDataContainer struct {
+	seq        int64
+	staleSeq   int64
+	missedSeqs int64
+	uuid       atomic.Value
+}
+
+func NewMetaDataContainer() *MetaDataContainer {
+	uuid := atomic.Value{}
+	uuid.Store(types.UUID{})
+	return &MetaDataContainer{
+		uuid: uuid,
+	}
+}
+
+func (mc *MetaDataContainer) SetUUID(uuid types.UUID) {
+	mc.uuid.Store(uuid)
+}
+
+func (mc MetaDataContainer) UUID() types.UUID {
+	return mc.uuid.Load().(types.UUID)
+}
+
+func (mc *MetaDataContainer) CASUUID(prev, new types.UUID) bool {
+	return mc.uuid.CompareAndSwap(prev, new)
+}
+
+func (mc *MetaDataContainer) SetSequence(seq int64) {
+	atomic.StoreInt64(&mc.seq, seq)
+}
+
+func (mc MetaDataContainer) Sequence() int64 {
+	return atomic.LoadInt64(&mc.seq)
+}
+
+func (mc *MetaDataContainer) ResetSequence() {
+	mc.SetSequence(0)
+}
+
+func (mc *MetaDataContainer) CASSequence(current, next int64) bool {
+	return atomic.CompareAndSwapInt64(&mc.seq, current, next)
+}
+
+func (mc *MetaDataContainer) SetStaleSequence(seq int64) {
+	atomic.StoreInt64(&mc.staleSeq, seq)
+}
+
+func (mc MetaDataContainer) StaleSequence() int64 {
+	return atomic.LoadInt64(&mc.staleSeq)
+}
+
+func (mc *MetaDataContainer) ResetStaleSequence() {
+	mc.SetStaleSequence(0)
+}
+
+func (mc *MetaDataContainer) CASStaleSequence(lastKnown, lastReceived int64) bool {
+	return atomic.CompareAndSwapInt64(&mc.staleSeq, lastKnown, lastReceived)
+}
+
+func (mc *MetaDataContainer) SetMissedSequenceCount(count int64) {
+	atomic.StoreInt64(&mc.missedSeqs, count)
+}
+
+func (mc MetaDataContainer) MissedSequenceCount() int64 {
+	return atomic.LoadInt64(&mc.missedSeqs)
+}
+
+func (mc *MetaDataContainer) AddAndGetMissedSequenceCount(count int64) int64 {
+	return atomic.AddInt64(&mc.missedSeqs, count)
+}
+
+// InvalidationMetaDataFetcher runs on Near Cache side.
+// An instance of this task is responsible for fetching of all Near Caches' remote metadata like last sequence numbers and partition UUIDs.
+// port of: com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher
+// port of: com.hazelcast.client.map.impl.nearcache.invalidation.ClientMapInvalidationMetaDataFetcher
+type InvalidationMetaDataFetcher struct {
+	cs         *cluster.Service
+	is         *invocation.Service
+	invFactory *cluster.ConnectionInvocationFactory
+	lg         ilogger.LogAdaptor
+}
+
+func NewInvalidationMetaDataFetcher(cs *cluster.Service, is *invocation.Service, invFactory *cluster.ConnectionInvocationFactory, lg ilogger.LogAdaptor) InvalidationMetaDataFetcher {
+	df := InvalidationMetaDataFetcher{
+		cs:         cs,
+		is:         is,
+		invFactory: invFactory,
+		lg:         lg,
+	}
+	return df
+}
+
+func (df InvalidationMetaDataFetcher) Init(ctx context.Context, handler RepairingHandler) error {
```

Comment:
Are you sure this function correctly ports ? com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher#init

Reply:
I've refactored this one at: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/4626aca1993000c79bfaaaf4acecca604b3d16e2